classes/local/store/s3/client.php

```diff
@@ -472,19 +462,30 @@ private function generate_presigned_url_cloudfront($contenthash, $headers = [],
         if ($nicefilename) {
             $key .= $this->get_nice_filename($headers);
         }
+        $resource = $this->config->cloudfrontresourcedomain . '/' . $key;
+        // This is the id of the Cloudfront key pair you generated.
+        $keypairid = $this->config->cloudfrontkeypairid;
 
-        $resourcekey = $this->config->cloudfrontresourcedomain . '/' . $key;
-        $signingparameters = [
-            'url' => $resourcekey,
-            'expires' => $expires,
-            'key_pair_id' => $this->config->cloudfrontkeypairid,
-            'private_key' => realpath($this->config->cloudfrontprivatekeypemfilepathname),
-        ];
-        $signedurlcannedpolicy = $client->getSignedUrl($signingparameters);
-        $signedurl = (string)$signedurlcannedpolicy;
-
-        /* $headers[] = 'Location:"' . $signedurl . '"'; // This may cause loss of headers (etag for example). */
-        return $signedurl;
+        $json = '{"Statement":[{"Resource":"' . $resource . '","Condition":{"DateLessThan":{"AWS:EpochTime":' . $expires . '}}}]}';
+
+        // Create the private key.
+        $key = openssl_get_privatekey($this->config->cloudfrontprivatekey);
```

Comment:
Just being my own devils advocate: in the auth_saml plugin we can accept certs either inline or as a url, we probably want similar flexibility here too, ie in some cases like this dev env we want to get up and running quickly inline but in prod we probably want to lock it down using a path which is forced. So I'm thinking of best of both worlds where we can accept either an inline cert or a path to a cert.

Should only be a couple lines, if the config item starts with '/' and it's a single line then treat it as a file path, if not treat as inline cert

Reply:
I'll implement in #254